src/ocl/convolutionocl.cpp

```diff
@@ -417,11 +470,6 @@ void ConvolutionDescriptor::ConvolutionForward(Handle& handle,
 {
     MIOPEN_LOG_I("algo = " << algo << ", workspace = " << workSpaceSize);
 
-    if(!(xDesc.IsPacked() && wDesc.IsPacked() && yDesc.IsPacked()))
-    {
-        MIOPEN_THROW(miopenStatusNotImplemented, "Only fully packed tensors are supported");
-    }
-
```

Comment:
@averinevg : What's the best way to go about finding these solvers? How about I update the test to not apply the check if the algorithms are "naive" or "ck" based?

Reply:
@amberhassaan You can find them all in `src/include/miopen/solver.hpp`. It is better to implement this check inside solvers' `IsApplicable()` because this is exactly the place where such checks should be made.

Reply:
A general function for checking can be added to `conv::ProblemDescription`

Reply:
Got it. However, due to shortage of time, I'd like an easier solution. It seems like we have many many solvers for convolution and updating all of them will take time.
@JehandadKhan , @junliume : thoughts? context: we need to disable the check above to allow non-packed tensors in convolution, but then the check is needed for all the solvers that don't support non-packed tensors, and those are like 99% of them.

Reply:
In `conv::ProblemDescription` create a new helper function which checks if all the tensor ( input. output, weights) are packed. Then in each solver's `IsApplicable` call it to enforce the condition. While I agree it is cumbersome but in the current implementation, that is how all of this is implemented. For example, when a new layout/data type is added, we have go through all solver and make sure the old solvers respect the new feature.

Reply:
ok, I agree with all of that. I just want to postpone this change to after our deadline.

Reply:
https://github.com/ROCmSoftwarePlatform/MIOpen/issues/2444 created to handle this as it's a more involved change. @junliume @JehandadKhan